docs/dev/action_collection.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Action Collection
+---
+
+# Action Collection Design
+
+The action collection tracks all actions taken by all Chef resources.  The resources can be in recipe code, as sub-resources of custom resources or
+they may be built "by hand".  Since the action collection hooks the events which are fired from the `run_action` method on Chef::Resource it does
+not matter how the resources were built (as long as they were correctly passed the Chef `run_context`).
+
+This is complementary to the resource collection itself which as incomplete statement of what might happen or has happened in the run since there are
```

Comment:
```suggestion
This is complementary to the resource collection itself which has an incomplete picture of what might happen or has happened in the run since there are
```